File: Source/JavaScript/model/layouts/FlowColumn.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { FlowContainer } from './FlowContainer';
import { FlowContainerKind } from './FlowContainerKind';

/**
 * A {@link FlowContainer} that arranges its children vertically.
 */
export interface FlowColumn extends FlowContainer {
    kind: FlowContainerKind.Column;
}

export const FlowColumnPropertyNames: (keyof FlowColumn)[] = ['kind'];

File: Source/DotNET/Model/Layouts/FlowColumn.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Scene.Model.Layouts;

/// <summary>
/// A <see cref="FlowContainer"/> that arranges its children vertically.
/// </summary>
public record FlowColumn : FlowContainer
{
    /// <inheritdoc/>
    public override FlowContainerKind Kind => FlowContainerKind.Column;
}
